compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -134,14 +134,23 @@ pub struct FrameInfo<'tcx> {
     pub lint_root: Option<hir::HirId>,
 }
 
+/// Unwind information.
+#[derive(Clone, Copy, Eq, PartialEq, Debug, HashStable)]
+pub enum StackPopUnwind {
+    /// The cleanup block.
+    Cleanup(Option<mir::BasicBlock>),
```

Comment:
Please explain the situation where the block is `None`.

Actually, it's probably better to "bubble up" the `Option` like this:
```rust
Cleanup(mir::BasicBlock),
Skip,
NotAllowed,
```

Reply:
> Please explain the situation where the block is `None`.

I did not found the explanation before, so I didn't write it here.
I guess I can write "no cleanup needs to be done".

> Actually, it's probably better to "bubble up" the `Option`

`Option<mir::BasicBlock>` is used in some other places, so I chose `Cleanup(Option<_>)` for simplicity, but I can change it.

Reply:
> I did not found the explanation before, so I didn't write it here.

That's fair. :) But I think we should use this opportunity to fix this.

>  I guess I can write "no cleanup needs to be done".

Yes, that's what this means: on unwind, just pop this stack frame without running any cleanup code.

> Option<mir::BasicBlock> is used in some other places, so I chose Cleanup(Option<_>) for simplicity, but I can change it.

Yeah and it's rather hacky, since for *return* blocks `None` means "returning is UB" but for *unwind* blocks `None` means "no cleanup needed". I think we should clean this up, since now "unwinding is UB" also becomes a possibility so things just become too confusing.